sdk/src/asset_handlers/font_io.rs

```diff
@@ -0,0 +1,882 @@
+/// significant position.
+///
+/// ### Examples
+///
+/// ```ignore
+/// // Cannot work as written because font_io is private.
+/// use c2pa::asset_handlers::font_io::u32_from_u16_pair;
+/// let full_word = u32_from_u16_pair(0x1234, 0x5678);
+/// assert_eq!(full_word, 0x12345678);
+/// ```
+#[allow(dead_code)]
+pub fn u32_from_u16_pair(hi: u16, lo: u16) -> Wrapping<u32> {
+    // TBD - Supposedly the bytemuck crate, already in this project, can help us
+    // with stuff like this.
+    Wrapping((hi as u32 * 65536) + lo as u32)
+}
+
+/// Gets the high-order the u32 from given u64
+///
+/// ### Parameters
+///
+/// - `big` - Unsigned 64-bit integer
+///
+/// ### Returns
+///
+/// u32 fullword extracted from the more-significant 32 bits of the given value.
+///
+/// ### Examples
+///
+/// ```ignore
+/// // Cannot work as written because font_io is private.
+/// use c2pa::asset_handlers::font_io::u32_from_u64_hi;
+/// let hi_word = u32_from_u64_hi(0x123456789abcdef0);
+/// assert_eq!(hi_word, 0x12345678);
+/// ```
+#[allow(dead_code)]
+pub fn u32_from_u64_hi(big: u64) -> Wrapping<u32> {
+    Wrapping(((big & 0xffffffff00000000) >> 32) as u32)
+}
+
+/// Gets the low-order u32 from the given u64
+///
+/// ### Parameters
+///
+/// - `big` - Unsigned 64-bit integer
+///
+/// ### Returns
+///
+/// u32 fullword extracted from the less-significant 32 bits of the given value.
+///
+/// ### Examples
+///
+/// ```ignore
+/// // Cannot work as written because font_io is private.
+/// use c2pa::asset_handlers::font_io::u32_from_u64_lo;
+/// let lo_word = u32_from_u64_lo(0x123456789abcdef0);
+/// assert_eq!(lo_word, 0x9abcdef0);
+/// ```
+#[allow(dead_code)]
+pub fn u32_from_u64_lo(big: u64) -> Wrapping<u32> {
+    Wrapping((big & 0x00000000ffffffff) as u32)
+}
+
+/// 'C2PA' font table as it appears in storage
+#[derive(Debug, Default)]
+#[repr(C, packed(4))] // As defined by the C2PA spec.
+#[allow(non_snake_case)] // As named by the C2PA spec.
+pub struct TableC2PARaw {
+    /// Specifies the major version of the C2PA font table.
+    pub majorVersion: u16,
+    /// Specifies the minor version of the C2PA font table.
+    pub minorVersion: u16,
+    /// Offset from the beginning of the C2PA font table to the section
+    /// containing a URI to the active manifest. If a URI is not provided a
+    /// NULL offset = 0x0000 should be used.
+    pub activeManifestUriOffset: u32,
+    /// Length of URI in bytes.
+    pub activeManifestUriLength: u16,
+    /// Reserved for future use.
+    pub reserved: u16,
+    /// Offset from the beginning of the C2PA font table to the section
+    /// containing a C2PA Manifest Store. If a Manifest Store is not provided a
+    /// NULL offset = 0x0000 should be used.
+    pub manifestStoreOffset: u32,
+    /// Length of the C2PA Manifest Store data in bytes.
+    pub manifestStoreLength: u32,
+}
+
+impl TableC2PARaw {
+    /// Constructs a new, empty, instance.
+    ///
+    /// ### Returns
+    ///
```

Comment:
nit: For `### Parameters` and `### Returns`, we have throughout the code:

```
### Returns
A new instance.
```

i.e., without the empty line between the heading the contents.

Reply:
Oh, I saw there was a mixture, so I settled on the style with the blank lines, remembering that's how our Markdown plugin likes it, and assuming that these comment blocks will someday show up in Markdown...

Reply:
ok, I haven't been in this project in a while. But when you go look at the other rust project you will see what I mentioned.

Reply:
Whatever's fine w/ me.